### DISCUSSION TITLE
the comments in the Type Narrowing sample page

### DISCUSSION THREAD

**Discussion by kuri-menu** [microsoft/pyright#8401] [Q&A]
```python
def func(val: float | str | complex, test: bool):
    reveal_type(val) # int | str | complex
```
The above is the code for the sample page, but I think `float` is correct instead of `int` as shown below.
```python
def func(val: float | str | complex, test: bool):
    reveal_type(val) # float | str | complex
```
https://microsoft.github.io/pyright/#/type-concepts-advanced?id=static-typing-advanced-topics

**Comment by erictraut** [comment]
Yep, you're correct. It's now fixed.
